lib/widgets/markdown_text.dart

```diff
@@ -8,6 +8,34 @@ import '../url_launcher.dart';
 import 'cached_network_image.dart';
 import 'fullscreenable_image.dart';
 
+/// Accepted formats:
+/// !community@server.com
+/// /c/community@server.com
+/// /m/community@server.com
+/// /u/username@server.com
+/// @username@server.com
+
+class LemmyLinkSyntax extends md.InlineSyntax {
+  // https://github.com/LemmyNet/lemmy-ui/blob/61255bf01a8d2acdbb77229838002bf8067ada70/src/shared/config.ts#L38
+  static const String _pattern =
+      r'(\/[cmu]\/|@|!)([a-zA-Z0-9._%+-]+)@([a-zA-Z0-9.-]+\.[a-zA-Z]{2,})';
+
+  LemmyLinkSyntax() : super(_pattern);
+
+  @override
+  bool onMatch(md.InlineParser parser, Match match) {
+    final modifier = match[1]!;
+    final name = match[2]!;
+    final url = match[3]!;
```

Comment:
if you merge the dart v3 pr first, you can do:
```suggestion
    final [modifier, name, url] = match.groups([1, 2, 3]);
```

Reply:
Fancy. I'll play around with the new syntax. gonna leave it as.isnhere cause it's visually more straightforward with the string tokenization.